src/main/java/teammates/ui/controller/ControllerServlet.java

```diff
@@ -126,7 +127,7 @@ public final void doPost(HttpServletRequest req, HttpServletResponse resp) throw
              * in GAE shutting down the instance.
              * Note that severe logs are sent by email automatically in the cron job auto/compileLogs.
              */
-            log.severe("Unexpected exception " + t.getMessage());
+            log.severe(TeammatesException.toStringWithStackTrace(t));
```

Comment:
Ok I have added that. However, there seems to be a failing test unrelated to what I have changed. I have added more details in the conversation.